### PR TITLE
Bg user story 9

### LINKED ISSUE #12: User Story 9, Admin Navigation Restrictions
As an admin
When I try to access any path that begins with the following, then I see a 404 error:
- '/merchant'
- '/cart'

### PR DESCRIPTION
**Pull request by brian-greeson**
## Authorized routes for Admin
Closes: #12 
### Created
- deny_admin method in cart controller.
   - renders 404 if admin user attempts any cart action 
- added deny_admin to before actions.

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -20,6 +22,10 @@ def remove_item
     redirect_to '/cart'
   end
 
+  def deny_admin
+    render file: "/public/404" if current_admin?
```
Comment (kelshadarby) [inline app/controllers/cart_controller.rb:28]: Nice naming here!

File: spec/features/admin/nav_spec.rb [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 require 'rails_helper'
 
-RSpec.describe 'As a Admin I have a navbar', type: :feature do
-  describe 'where I can click on' do
-    before(:all) do
-      @user1 = User.new(
+RSpec.describe 'As a Admin', type: :feature do
+  describe 'I have a navbar' do
+    it 'where I can click on' do
+      user1 = User.new(
```
Comment (kelshadarby) [inline spec/features/admin/nav_spec.rb:7]: Nice way to break this up

File: spec/features/admin/nav_spec.rb [hunk 3]
```diff
@@ -16,71 +16,80 @@
...
-    end
-
-    it 'Users' do
-      visit '/'
-      click_on 'Users'
-      expect(current_path).to eq(admin_users_path)
-    end
+  describe 'I do not have access' do
+    it '/merchant routes' do
+      user1 = User.new(
```
Comment (kelshadarby) [inline spec/features/admin/nav_spec.rb:56]: Nice naming here

File: spec/features/admin/nav_spec.rb [hunk 4]
```diff
@@ -16,71 +16,80 @@
...
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user1)
 
-    it 'Logout' do
-      visit '/'
-      click_on 'Logout'
-      expect(current_path).to eq(root_path)
-    end
-    after(:all) do
-      UserDetail.destroy_all
-      User.destroy_all
```
Comment (kelshadarby) [inline spec/features/admin/nav_spec.rb:83]: Nice teardown here!

File: app/controllers/cart_controller.rb [hunk 5]
```diff
@@ -20,6 +22,10 @@ def remove_item
     redirect_to '/cart'
   end
 
+  def deny_admin
```
Comment (kelshadarby) [inline app/controllers/cart_controller.rb:27]: Should this be a private method?

**Review by kelshadarby — APPROVED** [review]

File: app/controllers/cart_controller.rb (hunk 5 above)
Comment (brian-greeson) [reply app/controllers/cart_controller.rb:27]: We should decide a convention in our next group meeting

Comment (fredrondina) [reply app/controllers/cart_controller.rb:27]: I think its fine as is for now

**Review by fredrondina — APPROVED** [review]

**Pull request merged**